ui/coordinateTrainer/css/_coordinateTrainer.scss

```diff
@@ -0,0 +1,238 @@
+@import '../../puz/css/font';
+@import 'layout';
+@import 'zen';
+
+$mq-col1: $mq-not-medium;
+$mq-col2: $mq-medium;
+$mq-col3: $mq-x-large;
+
+#main-wrap {
+  --main-max-width: calc(100vh - #{$site-header-outer-height});
+
+  @include breakpoint($mq-col2) {
+    --main-max-width: auto;
+  }
+}
+
+.trainer {
+  grid-area: main;
+  display: grid;
+
+  .side {
+    @extend %zen;
+    grid-area: side;
+  }
+
+  .main-board {
+    grid-area: board;
+  }
+
+  .table {
+    grid-area: table;
+  }
+
+  .progress {
+    @extend %zen;
+    grid-area: progress;
+  }
+
+  grid-template-areas: 'board' 'progress' 'table' 'side';
+  grid-row-gap: $block-gap;
+
+  @include breakpoint($mq-col2) {
+    grid-template-columns: $col2-uniboard-width $block-gap $col2-uniboard-table;
+    grid-template-rows: fit-content(0);
+    grid-template-areas:
+      'board    . table'
+      'board    . side'
+      'progress . side'
+      '.        . side';
+  }
+
+  @include breakpoint($mq-col3) {
+    grid-template-areas:
+      'side . board    . table'
+      'side . progress . .'
+      'side . .        . .';
```

Comment:
Slight change here to fix this bug where the progress bar is always in a grid row beneath the side content:
![image](https://user-images.githubusercontent.com/12655228/157113354-25daeed7-1932-4e77-9f24-5fb64ee1e1a7.png)